### ISSUE TITLE
Invalid introspection response for fields with default value

### ISSUE THREAD

**Issue by MohamedSabthar** [ballerina-platform/ballerina-standard-library#3307]
**Description:**
Consider the following example
```ballerina
import ballerina/graphql;

public type Person record {
    string name;
    int age;
    string id;
};

service /graphql on new graphql:Listener(3000) {
    resource function get profie (string id = "1") returns Person {
        return {name: "sabthar", age: 23, id:"1"};
    }
}
```

Introspection for the above profile field returns following introspection response
```graphql
type Query {
  profie(id: String!): Person!
}
```
which is not valid. The response should be 
```graphql
type Query {
  profie(id: String = "1"): Person!
}
```

note the type of the id in the second response is not a  NON-NULL value. For more information [refer](https://spec.graphql.org/October2021/#sel-IALTHHFDABAB7EtqO)

**Steps to reproduce:**

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

**Comment by MohamedSabthar** [comment]
Due to the limitation in the compiler plugin, the default value is set to an empty string in the current ballerina graphql implementation. Some client programs could show errors when writing/creating a graphql document without the default value due to this current implentation. Apart from setting the default value to an empty string, the previous implementation is correct.

For the above service, the service should return the following type 
```graphql
type Query {
  profile(id: String! = "1"): Person!
}
```
Where ` id` can't take a null value and it has a default value along with it hence it is an optional parameter.

Simply put, an argument is required if and only if the argument type is non-null and does not have a default value. In every other case, the argument is optional.

**Comment by ThisaruGuruge** [comment]
To clarify more on this, there is a difference between `NON_NULL` arguments and required arguments. `NON_NULL` arguments don't necessarily have to be required arguments.
Apart from the default value shown in the introspection queries always being an empty string, the existing implementation works as expected. (Note that even though the shown default value is an empty string, the execution picks the correct/given default value). Therefore we should be reverting this change and fallback to the previous implementation. We will release a patch on top of the reversion.
